File: features.rb
FEATURES = {
  label: {
    name: 'label',
    description: 'When adding a "Mergeable" label to your pull request, it will pass the status check to green, otherwise red.'\
      'You can configure the label name with the query param `?name=AnotherLabel`',
    options: {
      name: 'mergeable'
    },
    github_events: ['pull_request']
  },
  reviews: {
    name: 'reviews',
    description: 'When 2 or more approvals are given to your pull request, it will pass the status check to green, otherwise red.'\
      'You can configure the minimum number of reviews with the query param `?minimum=1`',
    options: {
      minimum: 2
    },
    github_events: ['pull_request', 'pull_request_review']
  },
  'commit-msg': {
    name: 'commit-msg',
    description: 'When all commit messages of your pull request are respecting the conventionalcommits.org, it will pass the status check to green, otherwise red.',
    options: {},
    github_events: ['pull_request']
  }
}

File: lib/features.js
'use strict';

const FEATURES = {
    label: {
        name: 'label',
        description: 'Require pull request to be flagged with a dedicated label before merge.',
        options: {
            name: 'mergeable',
            branch: '',
        },
        github_events: ['pull_request'],
        img: 'label.png',
    },
    reviews: {
        name: 'reviews',
        description: 'Require pull request with required number of approving reviews and no changes requested.',
        options: {
            minimum: 2,
            branch: '',
        },
        github_events: ['pull_request', 'pull_request_review'],
        img: 'review.png',
    },
    'commit-msg': {
        name: 'commit-msg',
        description: 'Require pull request to have commit messages respecting the conventionalcommits.org.',
        options: {
            branch: '',
        },
        github_events: ['pull_request'],
        img: 'commit-msg.png',
    },
    branch: {
        name: 'branch',
        description: 'Require request branch name match with the given pattern.',
        options: {
            pattern: '*',
        },
        github_events: ['pull_request'],
        img: 'branch.png',
    },
};

export default FEATURES;
